File: pingport.rb
require 'socket'    
include Socket::Constants            
def ping_port(address, port)           
  begin    
    socket = Socket.new(AF_INET, SOCK_STREAM, 0)          
    socket.connect(Socket.pack_sockaddr_in(port, address))     
    puts "Port:#{port} is Opened~"    
    socket.close     
  rescue 
    puts "Port:#{port} is not available!\n"   
  end            
end
ping_port("127.0.0.1",8080)#input here  

File: pingport.py
import os
import socket
def IsAvailable(ip,port):
    s = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
    try:
        s.connect((ip,int(port)))
        s.shutdown(2)
        # print '%d available~' % port #py2
        print ('%d available~' % port) #py3
        return True
    except:
        # print '%d is busy!' % port #py2
        print ('%d is busy!' % port) #py3
        return False
if __name__ == '__main__':
    IsAvailable('127.0.0.1',8080)#input here
